core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -223,8 +222,10 @@ private[spark] abstract class MemoryManager(
       case MemoryMode.OFF_HEAP => offHeapExecutionMemoryPool.poolSize
     }
     val size = ByteArrayMethods.nextPowerOf2(maxTungstenMemory / cores / safetyFactor)
-    val default = math.min(maxPageSize, math.max(minPageSize, size))
-    conf.getSizeAsBytes("spark.buffer.pageSize", default)
+    val maxPageSize = math.min(64L * minPageSize, math.max(minPageSize, size))
+    val userSetting = conf.getSizeAsBytes("spark.buffer.pageSize")
+    // In case of too large page size.
+    math.min(userSetting, maxPageSize)
   }
```

Comment:
@JoshRosen The `SparkEnv.memoryManager.pageSizeBytes` returns `Long`, if we reuse it as chunk size, there is still underlying integer overflow, isn't it? Here, I restricted the upper limit of page size in case of too large.